File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    # 노란색,갈색개수에 맞는 격자크기 구하기
    for i in range(3,20000):
        for j in range(3,200000):
            if i*j > brown+yellow:
                break
            if i*j == brown+yellow and (i-2)*(j-2) == yellow:
                return j,i

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
  const TOTAL = brown + yellow;

  for (let height = 3; height <= Math.sqrt(TOTAL); height++) {
    const remainder = TOTAL % height;

    if (remainder !== 0) continue;

    const width = TOTAL / height;
    if ((width - 2) * (height - 2) === yellow) {
      return [width, height];
    }
  }
}